docs/v1/P4Runtime-Spec.mdk

```diff
@@ -2630,52 +2634,53 @@ following semantics.
 
 * `INSERT`: Add a new group entry bound to a set of existing action profile
   members. `The group_id` must be different from ids of already programmed
-  groups for that selector. P4Runtime does not limit the number of groups,
-  however, such limits may be imposed out-of-band by the target. The value of
-  `max_size` should not exceed the static maximum size defined for the selector
-  in P4Info, otherwise an `INVALID_ARGUMENT` error is returned. If the client
-  does not set `max_size`, the default value (0) implies that the
-  statically-defined maximum size should be used for this group.
+  groups for that selector, or the server must return an `ALREADY_EXISTS` error
+  code. P4Runtime does not explicitly limit the number of groups, however, such
+  limits may be imposed out-of-band by the target. The value of `max_size`
+  should not exceed the static maximum size defined for the selector in P4Info,
+  otherwise a `RESOURCE_EXHAUSTED` error should be returned. If the client does
+  not set `max_size`, the default value (0) implies that the statically-defined
+  maximum size should be used for this group.
 - `MODIFY`: Modify the member set specification of an existing group entry. An
-  entry with the `group_id` must exist. All members specified in the group entry
-  must exist in the selector. The value of `max_size` must be identical to the
-  value used when inserting the group, otherwise an `INVALID_ARGUMENT` error is
-  returned.
-- `DELETE`: Delete the group entry and deallocate the `group_id`. The group
-  should not be referenced in the table action of any table entry.
+  entry with the `group_id` must exist, or the server must return
+  `NOT_FOUND`. All members specified in the group entry must exist in the
+  selector, or the server must return `NOT_FOUND`. The value of `max_size` must
```

Comment:
A client receiving a NOT_FOUND response has no way to know whether it's because the group_id does not exist or a member does not exist. Would it make sense to differentiate them?

Reply:
NOT_FOUND seems like the most appropriate error code in both of these situations, however we could switch to FAILED_PRECONDITION for the second case (one of the members do not exist). Is there value in being able to separate these cases in a programmatic fashion in the client, or should we just rely on the error message?

Reply:
I haven't checked, but I would guess there are other request messages where the same error response indicates one of multiple possible error conditions.  OK to leave as is.